test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -2607,5 +2607,57 @@ def test_annotated_mcx(self):
         self.assertEqual(Operator(qc), Operator(qct))
 
 
+@ddt
+class TestPauliEvolutionSynthesisPlugins(QiskitTestCase):
+    """Tests related to plugins for PauliEvolutionGate."""
+
+    def test_supported_names(self):
+        """Test that "default" and "rustiq" plugins do exist."""
+        supported_plugin_names = high_level_synthesis_plugin_names("PauliEvolution")
+        self.assertIn("default", supported_plugin_names)
+        self.assertIn("rustiq", supported_plugin_names)
+
+    @data("default", "rustiq")
+    def test_correctness(self, plugin_name):
+        """Test that all plugins return a correct Operator"""
+        op = SparsePauliOp(["XXX", "YYY", "IZZ", "XZY"], coeffs=[1, 2, 3, 4])
```

Comment:
Could we also add a check for `preserve_order=False`?

Reply:
I don't know how to add a `correctness` test: since with `preserve_order=False` the operator might be completely different, but I have added a test that checks that the argument is applied.